scripts/system_performance/controller/method_mapping.py

```diff
@@ -0,0 +1,218 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Handles User input (N instruments, N methods and N additional method arguments), mapping
+user specified methods formatted as strings to their corresponding method in controller applying
+any additional arguments specified.
+
+:returns a dictionary of dictionaries containing statistics per instrument, per method
+"""
+
+# Dependencies
+import logging
+
+from scripts.system_performance.controller.statistics_computation import QueryHandler
+from scripts.system_performance.models import query_argument_constructor
+
+
+class MethodSelectorConfigurator(object):
+    """Class containing logic to call N methods specified by user for N instruments + any additional
+     method arguments specified"""
+
+    def __init__(self):
+        # create_method_mappings
+        pass
+
+    @staticmethod
+    def create_method_mappings():
+        """A dictionary to map input, user specified methods to return equivalent method to be called
+        TODO: Create execution_time_average and run_frequency_average methods"""
+
+        return {'missing_run_numbers_report': QueryHandler().missing_run_numbers_report,
+                'execution_times': QueryHandler().execution_times,
+                # 'execution_time_average': self.execution_time_average, TODO create method
+                'run_frequency': QueryHandler().run_frequency,
+                # 'run_frequency_average': self.run_frequency_average _ TODO create method
+               }
+
+    def method_call(self, method_name, method_args):
+        """Calls user specified method and returns statistics for a given instrument"""
+        # Check input is in mapping and place method N output in instrument_dict to return
+
+        try:
+            method_output = self.create_method_mappings()[method_name](**method_args)
+        except KeyError:
+            logging.warn("Invalid Input - method '%s' does not exist try -help "
+                          "to look at existing methods and arguments", method_name)
+            method_output = None
+
+        except TypeError:
+            logging.warn("Invalid Input invalid addiitonal arguments entered "
+                         "for method: '%s'", method_name)
+            method_output = None
+        return method_output
+
+    @staticmethod
+    def get_instrument_models():
```

Comment:
perhaps a level too high of abstraction, instead we could just let functions call `query_argument_constructor.get_list_of_instruments()`. Unless there is a reason we can't do this?